fix(chat): add localStorage save and clear functionality

decodewidwaiz opened the pull request:
### Summary

- Chat messages are now saved to localStorage.
- A trash icon was added to clear the chat history.
- On clicking the icon, localStorage is also cleared along with the UI.

### Changes Made

- Updated chat logic to persist data via localStorage
- Added delete (🗑️) button to UI
- Attached event listener to clear both UI and storage

### Screenshots (if UI changed)

<img width="1911" height="965" alt="image" src="https://github.com/user-attachments/assets/a6b6f455-1be2-4548-8406-195238c1479e" />

<upload if applicable>

### Related Issues

Closes #157 (if there's a linked GitHub issue)

---

RishuKumarCodes commented:
why are you using emoji for the icon? use react-native svg for icon, make icon red on hover. make sure on opening the chat, it shows the end of chat, not the Top of the message. test through all edge cases.

decodewidwaiz commented:
i have made changes: 
1. removed bin emoji to bin icon.
2. checked all the test cases. earlier it get vanished on refresh, fixed that issue.
3. chat will stay even if we refresh.
4. bin icon will properly working.
<img width="1905" height="959" alt="image" src="https://github.com/user-attachments/assets/462f6ef4-8b05-4467-9a67-3195707b5f76" />

decodewidwaiz commented:
made changes
<img width="443" height="644" alt="image" src="https://github.com/user-attachments/assets/aef86cbc-cc7d-4c05-b6ff-c0529fce57c2" />
